src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -202,6 +204,58 @@ private static Uri GetUriFromString(string value)
             return null;
         }
 
+        /// <summary>
+        /// This function is used to parse WinINet Proxy strings. The strings are a semicolon
+        /// or whitespace separated list, with each entry in the following format:
+        /// ([<scheme>=][<scheme>"://"]<server>[":"<port>])
+        /// </summary>
+        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy )
+        {
+            secureProxy = null;
+            insecureProxy = null;
+            if (string.IsNullOrEmpty(value))
+            {
+                return;
+            }
+
+            int idx = value.IndexOf("http://");
+            if (idx >= 0)
+            {
+                int proxyLength = GetProxySubstringLength(value, idx);
+                Uri.TryCreate(value.Substring(idx, proxyLength) , UriKind.Absolute, out insecureProxy);
+            }
+
+            idx = value.IndexOf("http=");
+            if (idx >= 0)
+            {
+                idx += 5; // Skip "http=" so we can replace it with "http://"
+                int proxyLength = GetProxySubstringLength(value, idx);
+                Uri.TryCreate("http://" + value.Substring(idx, proxyLength) , UriKind.Absolute, out insecureProxy);
+            }
```

Comment:
Should this block be surrounded by:
```C#
if (insecureProxy == null)
{
    ...
}
```
so that we only do it if the first attempt for http:// failed?  Seems like that could be useful as a tiny perf benefit, but more so to avoid a situation where you have:
```
http=http://server
```
as the entry, at which point it looks like this would end up setting `insecureProxy` to null, as it would try to parse a url out of `http://http://server`.

Reply:
Good catch. I'll add that check and a test case.